From 1bfd7921284e84ea92bc3aebd0b70d974d972112 Mon Sep 17 00:00:00 2001
From: Stanislav Kholmanskikh <stanislav.kholmanskikh@bell-sw.com>
Date: Fri, 20 May 2022 10:47:14 +0300
Subject: [PATCH] dracut/alpaquita-iso: enable getty on the serial port

This is useful for install automation.

The 'once' is used to avoid filling the logs with port open
error messages when the system doesn't have a serial port.
---
 modules.d/98alpaquita-iso/mount-tmpfs.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/98alpaquita-iso/mount-tmpfs.sh b/modules.d/98alpaquita-iso/mount-tmpfs.sh
index cfb2f81..f5bd785 100755
--- a/modules.d/98alpaquita-iso/mount-tmpfs.sh
+++ b/modules.d/98alpaquita-iso/mount-tmpfs.sh
@@ -90,3 +90,7 @@ EOD
 # Start the installer on boot
 sed -i -r 's,^(#?(tty1::respawn:).*),\2/usr/bin/alpaquita-installer -i -d,' \
 	"$NEWROOT/etc/inittab"
+
+# Enable getty on the serial port
+sed -i -r 's,^(#?(ttyS0)::respawn:.*),\2::once:/sbin/getty -L ttyS0 115200 vt100,' \
+	"$NEWROOT/etc/inittab"
-- 
2.34.1

